Python/ast_unparse.c

```diff
@@ -0,0 +1,1040 @@
+            return ret;
+    }
+    return omit_parens ? 0 : append_charp(writer, ")");
+}
+
+static int
+append_ast_await(_PyUnicodeWriter *writer, expr_ty e, bool omit_parens)
+{
+    int ret;
+
+    ret = omit_parens ? 0 : append_charp(writer, "(");
+    if (ret == -1)
+        return ret;
+
+    ret = append_charp(writer, e->v.Await.value ? "await " : "await");
+    if (ret == -1)
+        return ret;
+
+    if (e->v.Yield.value)
+    {
+        ret = append_ast_expr(writer, e->v.Await.value, false, false);
+        if (ret == -1)
+            return ret;
+    }
+    return omit_parens ? 0 : append_charp(writer, ")");
+}
+
+static int
+append_ast_str(_PyUnicodeWriter *writer, expr_ty e, bool omit_string_brackets)
+{
+    PyObject *s =  e->v.Str.s;
+    if (omit_string_brackets)
+        return _PyUnicodeWriter_WriteStr(writer, s);
+
+    return append_repr(writer, s);
+}
+
+static int
+append_ast_expr(_PyUnicodeWriter *writer, expr_ty e, bool omit_parens,
+    bool omit_string_brackets)
+{
+    switch (e->kind) {
+    case BoolOp_kind:
+        return append_ast_boolop(writer, e, omit_parens);
+    case BinOp_kind:
+        return append_ast_binop(writer, e, omit_parens);
+    case UnaryOp_kind:
+        return append_ast_unaryop(writer, e, omit_parens);
+    case Lambda_kind:
+        return append_ast_lambda(writer, e, omit_parens);
+    case IfExp_kind:
+        return append_ast_ifexp(writer, e, omit_parens);
+    case Dict_kind:
+        return append_ast_dict(writer, e);
+    case Set_kind:
+        return append_ast_set(writer, e);
+    case GeneratorExp_kind:
+        return append_ast_genexp(writer, e, omit_parens);
+    case ListComp_kind:
+        return append_ast_listcomp(writer, e);
+    case SetComp_kind:
+        return append_ast_setcomp(writer, e);
+    case DictComp_kind:
+        return append_ast_dictcomp(writer, e);
+    case Yield_kind:
+        return append_ast_yield(writer, e, omit_parens);
+    case YieldFrom_kind:
+        return append_ast_yield_from(writer, e, omit_parens);
+    case Await_kind:
+        return append_ast_await(writer, e, omit_parens);
+    case Compare_kind:
+        return append_ast_compare(writer, e, omit_parens);
+    case Call_kind:
+        return append_ast_call(writer, e);
+    case Constant_kind:
+        return append_repr(writer, e->v.Constant.value);
+    case Num_kind:
+        return append_repr(writer, e->v.Num.n);
+    case Str_kind:
+        return append_ast_str(writer, e, omit_string_brackets);
+    case JoinedStr_kind:
+        PyErr_SetString(PyExc_SystemError,
+                        "f-string support in annotations not implemented yet");
+        return -1;
+    case FormattedValue_kind:
+        PyErr_SetString(PyExc_SystemError,
+                        "f-string support in annotations not implemented yet");
+        return -1;
+    case Bytes_kind:
+        return append_repr(writer, e->v.Bytes.s);
+    case Ellipsis_kind:
+        return append_charp(writer, "...");
```

Comment:
`Constant_kind` can be the Ellipsis too, in which case it outputs `Ellipsis` instead of `...`.

Reply:
Well, yes, because apparently the user wrote "Ellipsis" and not "...". Do you think we should convert to "..." in this case, too?

Reply:
FWIW:
```
>>> Ellipsis is ...
True
``` 
so this looks relatively low-pri.